File: titanium-front/src/app/model/inventory-element.ts
import {Batch} from './batch';

export class InventoryElement {

  name: string;
  more: string;
  perishable: boolean;
  minimum: number;
  stock: number;
  toOrder: number;
  expirationDate: Date;
  location: string;
  tags: String[];
  batch: Batch;
  uptodate: boolean;
  _id: string;

  constructor(name: string, more: string, perishable: boolean, minimum: number, stock: number, toOrder: number, expirationDate: Date, location: string, tags: String[], batch: Batch, uptodate: boolean, id: string = null) {
    this.name = name;
    this.more = more;
    this.perishable = perishable;
    this.minimum = minimum;
    this.stock = stock;
    this.toOrder = toOrder;
    this.expirationDate = expirationDate;
    this.location = location;
    this.tags = tags;
    this.batch = batch;
    this.uptodate = uptodate;
    this._id = id;
  }

}

File: titanium-services/src/main/kotlin/com/xrouge/mot/titanium/model/InventoryElement.kt
package com.xrouge.mot.titanium.model

import org.joda.time.LocalDate


data class InventoryElement(val name: String,
                            val more: String?,
                            val perishable: Boolean?,
                            val minimum: Int,
                            val stock: Int?,
                            val expirationDate: LocalDate?,
                            val location: ClosetLocation,
                            val tags: List<String>,
                            val batch: Batch,
                            var uptodate: Boolean,
                            val _id: String? = null) {

    fun toElement(): Element {
        return Element(name, more, perishable, minimum, stock!!, expirationDate, location, tags, batch)
    }
}